drivers/sensor/bmi08x/bmi08x_gyro.c

```diff
@@ -0,0 +1,525 @@
+	const struct spi_buf tx_buf[2] = {{.buf = &reg, .len = 1}, {.buf = data, .len = length}};
+	const struct spi_buf_set tx = {.buffers = tx_buf, .count = write ? 2 : 1};
+
+	if (!write) {
+		uint16_t dummy;
+		const struct spi_buf rx_buf[2] = {{.buf = &dummy, .len = 1},
+						  {.buf = data, .len = length}};
+		const struct spi_buf_set rx = {.buffers = rx_buf, .count = 2};
+
+		return spi_transceive_dt(&bmi08x->bus.spi, &tx, &rx);
+	}
+
+	return spi_write_dt(&bmi08x->bus.spi, &tx);
+}
+
+static int bmi08x_bus_check_spi(const union bmi08x_bus *bus)
+{
+	return spi_is_ready(&bus->spi) ? 0 : -ENODEV;
+}
+
+static const struct bmi08x_gyro_bus_io bmi08x_spi_api = {
+	.check = bmi08x_bus_check_spi,
+	.transceive = bmi08x_gyro_transceive_spi,
+};
+
+#endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(spi) */
+
+static inline int bmi08x_bus_check(const struct device *dev)
+{
+	const struct bmi08x_gyro_config *config = dev->config;
+
+	return config->api->check(&config->bus);
+}
+
+static int bmi08x_gyro_transceive(const struct device *dev, uint8_t reg, bool write, void *data,
+				  size_t length)
+{
+	const struct bmi08x_gyro_config *cfg = dev->config;
+
+	return cfg->api->transceive(dev, reg, write, data, length);
+}
+
+int bmi08x_gyro_read(const struct device *dev, uint8_t reg_addr, uint8_t *data, uint8_t len)
+{
+	return bmi08x_gyro_transceive(dev, reg_addr | BIT(7), false, data, len);
+}
+
+int bmi08x_gyro_byte_read(const struct device *dev, uint8_t reg_addr, uint8_t *byte)
+{
+	return bmi08x_gyro_transceive(dev, reg_addr | BIT(7), false, byte, 1);
+}
+
+int bmi08x_gyro_byte_write(const struct device *dev, uint8_t reg_addr, uint8_t byte)
+{
+	return bmi08x_gyro_transceive(dev, reg_addr & 0x7F, true, &byte, 1);
+}
+
+int bmi08x_gyro_word_write(const struct device *dev, uint8_t reg_addr, uint16_t word)
+{
+	uint8_t tx_word[2] = {(uint8_t)(word & 0xff), (uint8_t)(word >> 8)};
+
+	return bmi08x_gyro_transceive(dev, reg_addr & 0x7F, true, tx_word, 2);
+}
+
+int bmi08x_gyro_reg_field_update(const struct device *dev, uint8_t reg_addr, uint8_t pos,
+				 uint8_t mask, uint8_t val)
+{
+	uint8_t old_val;
+
+	if (bmi08x_gyro_byte_read(dev, reg_addr, &old_val) < 0) {
+		return -EIO;
+	}
+
+	return bmi08x_gyro_byte_write(dev, reg_addr, (old_val & ~mask) | ((val << pos) & mask));
+}
+
+#if defined(CONFIG_BMI08X_GYRO_ODR_RUNTIME)
+/*
+ * Output data rate map with allowed frequencies:
+ * freq = freq_int + freq_milli / 1000
+ *
+ * Since we don't need a finer frequency resolution than milliHz, use uint16_t
+ * to save some flash.
+ */
+static const struct {
+	uint16_t freq_int;
+	uint16_t freq_milli; /* User should convert to uHz before setting the
+			      * SENSOR_ATTR_SAMPLING_FREQUENCY attribute.
+			      */
+} bmi08x_odr_map[] = {
+	{0, 0},	 {0, 780}, {1, 562}, {3, 120}, {6, 250}, {12, 500}, {25, 0},
+	{50, 0}, {100, 0}, {200, 0}, {400, 0}, {800, 0}, {1600, 0}, {3200, 0},
+};
+
+static int bmi08x_freq_to_odr_val(uint16_t freq_int, uint16_t freq_milli)
```

Comment:
Prompt for repeated definition at compile time
in bmi08x_accel.c it has a same function name

Reply:
I think this part is public and should be separated to bmi08x.c

Reply:
yeah, and there's some other functions around it that ought to be common between the two. I'll get to moving it